### ISSUE TITLE
Description of each algorithm

### ISSUE THREAD

**Issue by the-ethan-hunt** [OpenGenus/cosmos#568]
It is also good to include a `readme.md` file in each algorithm to explain how the algorithm functions.

**Comment by ashutosh97** [comment]
I'd like to work on this! Should i include the `readme.md` file in the folder which contains the cpp file ?

**Comment by AdiChat** [comment]
@ashutosh97 Yes, the `readme.md` file exists at the same level as the code 👍

**Comment by ashutosh97** [comment]
Created a `readme.md` file for Segment Trees. Please review and merge - 
 [https://github.com/OpenGenus/cosmos/pull/591](url)

**Comment by SaiSakethAluru** [comment]
I would like to work on this.

**Comment by SaiSakethAluru** [comment]
PR #629 readme.md for knapsack problem. Please review and merge
